wire-ios-request-strategy/Sources/Object Syncs/Helpers/MLSMessage.swift

```diff
@@ -20,14 +20,31 @@ import Foundation
 
 /// A message that can be sent in an mls group.
 
-public protocol MLSMessage: OTREntity, MLSEncryptedPayloadGenerator {}
+public protocol MLSMessage: OTREntity, MLSEncryptedPayloadGenerator {
+
+    /// Messages can expire, e.g. if network conditions are too slow to send.
+    var shouldExpire: Bool { get }
+
+    /// Sets the expiration date with the default time interval and returns the date.
+    func setExpirationDate()
+}
 
 extension ZMClientMessage: MLSMessage {}
 
 extension ZMAssetClientMessage: MLSMessage {}
 
 extension GenericMessageEntity: MLSMessage {
+
+    // Just required for protocol conformance.
+    public var shouldExpire: Bool { false }
+
     public func encryptForTransport(using encrypt: (Data) async throws -> Data) async throws -> Data {
         try await message.encryptForTransport(using: encrypt)
     }
+
+    public func setExpirationDate() {
+        // Just required for protocol conformance.
+        // Generic messages are used as underlying messages in proteus and mls,
+        // so they don't need to have their own expiration date.
+    }
```

Comment:
At least this is @johnxnguyen 's and my assumption... please add knowledge if you know more :)